bundles/org.openhab.binding.sunsynk/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,179 @@
+			<channel id="inverter-dc-temperature" typeId="system.indoor-temperature"/>
+			<channel id="inverter-grid-power" typeId="system.electric-power"/>
+			<channel id="inverter-grid-voltage" typeId="system.electric-voltage"/>
+			<channel id="inverter-grid-current" typeId="system.electric-current"/>
+			<channel id="inverter-solar-energy-today" typeId="system.electric-energy"/>
+			<channel id="inverter-solar-energy-total" typeId="system.electric-energy"/>
+			<channel id="inverter-solar-power-now" typeId="system.electric-current"/>
+			<channel id="inverter-control-timer" typeId="control-timer"/>
+			<channel id="inverter-control-energy-pattern" typeId="control-pattern"/>
+			<channel id="inverter-control-work-mode" typeId="control-work-mode"/>
+		</channels>
+
+		<properties>
+			<property name="vendor">SunSynk</property>
+		</properties>
+		<representation-property>serialnumber</representation-property>
+
+		<config-description>
+			<parameter name="serialnumber" type="text" required="true">
+				<label>Inverter Serial Number</label>
+				<description>Serial number of the inverter from Sun Synk Connect.</description>
+			</parameter>
+			<parameter name="alias" type="text" required="true">
+				<label>Inverter Name</label>
+				<description>Name of inverter from Sun Synk Connect.</description>
+			</parameter>
+			<parameter name="refresh" type="integer" required="true">
+				<label>Refresh Interval</label>
+				<description>Time in seconds to retrieve status from Sun Synk Connect services. Greater than 60 seconds.</description>
+				<default>60</default>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="grid-time" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Grid Charge Time</label>
+		<description>"Grid charging interval start time. Use 24hr time 00:00 to 23:55 are valid in 5 min increments with
+			leading zero
+			on the hour and minute. Interval times must be in ascending order."</description>
+		<state readOnly="false"/> <!--- pattern="%1$tH:%1$tM"/> -->
+	</channel-type>
+
+	<channel-type id="grid-charge" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Grid Charge</label>
+		<description>Charge the inverter battery from the grid during this interval.</description>
+		<state readOnly="false" pattern="%s"/>
+	</channel-type>
+
+	<channel-type id="gen-charge" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Generator (AUX) Charge</label>
+		<description>Charge the inverter battery from the generator (auxilary) during this interval.</description>
+		<state readOnly="false" pattern="%s"/>
+	</channel-type>
+
+	<channel-type id="battery-target" advanced="true">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Target Charge Level</label>
+		<description>The desired charge level of the battery (%).</description>
+		<category>Battery</category>
+		<state min="0" max="100" step="1" pattern="%d" readOnly="false"/>
```

Comment:
Unit should also be set for percentage items. Either via uihint, or in the pattern. I suggested the lather:
```suggestion
		<state min="0" max="100" step="1" pattern="%d %%" readOnly="false"/>
```

Reply:
okay